### PR TITLE
Feature: Classify cookies by its use case and origin 

### LINKED ISSUE #3: Classify observed cookies across different categories
## Feature Request

**Description:**
Capability of categorizing cooking across different dimensions:

- Origin
- Lifespan
- Purpose
  - Functional (also known as technical, essential or strictly necessary)
  - Preferences (also known as functionality)
  - Analytics (also known as performance or statistics)
  - Marketing (also known as tracking or social media)

**User Story:**
As a developer/website owner/web user, I want to get an understanding of the types of cookies being set by my site and/or my browsing activities.

**Acceptance Criteria:**
- [ ] All observed cookies are checked for presence in the cookie DB and properly labeled when displayed in the cookie list.

### PR DESCRIPTION
**Pull request by ayushnirwal**
## Description

- Adds script to download [open-cookie-databse](https://raw.githubusercontent.com/rtCamp/Open-Cookie-Database/add-wordpress-cookies/open-cookie-database.csv) and formats it as a JSON file ( called cookieDB ).
- Adds category tags on the basis of 
  - Origin - Third party/First Party.
  - Use case - Functional/Preferences/Analytics/Marketing/Unknown.
- Adds interception for cookies sent in responses.

## Relevant Technical Choices

- Update `cookieData` type to support new `analytics` value.
- Update webpack config to bundle cookie DB file.
- Update manifest file to add cookie DB file as an extension asset.
- Update `ListItem` component to display origin and use case classification tags.  
- 4 Utility functions added.
  - `isFirstParty`- util function for origin based classification.
  - `parseRequestCookieHeader` -  for parsing request cookies.
  - `fetchCookieDictionary` - for fetching cookieDB.
  - `findAnalyticsMatch` - for finding a cookie's analytics in the cookieDB.
- Add `npm run cookie-db:update` command to fetch, format and create a JSON file (CookieDB) 

## Testing Instructions

- Uninstall any previous versions of the extension if installed.
- Follow instructions in the README.MD to build and load the extension.
- Open a new tab and go to any website 
- Open the devtools panel and go to "Privacy sandbox" tab.
- A scrollable list of cookies should be visible that were used in the network requests generated by the tab.

## Screenshot/Screencast
Screenshot 2023-06-23 at 10 09 45 AM

---

Fixes #3

### CODE REVIEW

File: data/open-cookie-database.json
Comment (sayedtaqui) [inline data/open-cookie-database.json:1]: Lets move this to a new `third_party` folder in the root.

File: scripts/update-cookie-db.cjs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+const path = require('path');
+const csvtojson = require('csvtojson');
+const process = require('process');
+const fetch = require('node-fetch');
+const targetDIR = path.resolve(__dirname, '..', 'third_party/data');
+
+// @todo Update to https://raw.githubusercontent.com/jkwakman/Open-Cookie-Database/master/open-cookie-database.csv after
+// https://github.com/jkwakman/Open-Cookie-Database/pull/32 is merged.
+const requestURL =
+  'https://raw.githubusercontent.com/rtCamp/Open-Cookie-Database/add-wordpress-cookies/open-cookie-database.csv';
```
Comment (sayedtaqui) [inline scripts/update-cookie-db.cjs:30]: @ayushnirwal Temporarily updated the url to fetch the updated cookie database until https://github.com/jkwakman/Open-Cookie-Database/pull/32 is merged

File: packages/extension/src/manifest.json [hunk 2]
```diff
@@ -18,5 +18,11 @@
   },
   "action": {
     "default_icon": "icons/icon-48.png"
-  }
+  },
+  "web_accessible_resources": [
+    {
+      "resources": ["data/*.json"],
```
Comment (sayedtaqui) [inline packages/extension/src/manifest.json:24]: Shouldn't this be `third_party/data/*.json` now?

File: packages/extension/src/utils/fetchCookieDictionary.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+export type CookieAnalytics = {
+  platform: string;
+  category: string;
+  name: string;
+  domain: string;
+  description: string;
+  retention: string;
+  dataController: string;
+  GDPRUrl: string;
```
Comment (sayedtaqui) [inline packages/extension/src/utils/fetchCookieDictionary.ts:25]:
```suggestion
  gdprUrl: string;
```

File: packages/extension/src/worker/findAnalyticsMatch.ts [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * Internal dependencies.
+ */
+import type {
+  CookieAnalytics,
+  CookieDatabase,
+} from '../utils/fetchCookieDictionary';
+
+/**
+ *
```
Comment (sayedtaqui) [inline packages/extension/src/worker/findAnalyticsMatch.ts:26]: Lets add description

File: packages/extension/src/worker/findAnalyticsMatch.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+const wildTest = (wildcard: string, str: string): boolean => {
+  const regExp = wildcard.replace(/[.+^${}()|[\]\\]/g, '\\$&'); // regexp escape
+  const result = new RegExp(
+    `^${regExp.replace(/\*/g, '.*').replace(/\?/g, '.')}$`,
+    'i'
+  );
+  return result.test(str); // remove last 'i' above to have case sensitive
+};
+
+const findAnalyticsMatch = (
```
Comment (sayedtaqui) [inline packages/extension/src/worker/findAnalyticsMatch.ts:47]: Please add description.

**Review by sayedtaqui — CHANGES_REQUESTED** [review]

File: packages/extension/src/worker/findAnalyticsMatch.ts
[the remark this answers is not in the document]
Comment (ayushnirwal) [reply packages/extension/src/worker/findAnalyticsMatch.ts:1]: tests for 'parseRequestCookieHeader' and 'parseResponseCookieHeader' covers their testing.

File: packages/extension/src/worker/findAnalyticsMatch.ts (hunk 5 above)
Comment (ayushnirwal) [reply packages/extension/src/worker/findAnalyticsMatch.ts:47]: Done in https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/25/commits/6c73e0b2e467aa0e280ce72e6b30272f76feb034

**Review by sayedtaqui — APPROVED** [review]

**Pull request merged**